monkey/infection_monkey/master/control_channel.py

```diff
@@ -36,11 +37,14 @@ def should_agent_stop(self) -> bool:
 
             response = json.loads(response.content.decode())
```

Comment:
Should we also `response.raise_for_status()`? What about in `get_credentials_for_propagation()`?